etna/datasets/tsdataset.py

```diff
@@ -857,6 +868,45 @@ def train_test_split(
 
         return train, test
 
+    def update_columns_from_pandas(self, df: pd.DataFrame, regressors: Optional[List[str]] = None):
+        """Update the dataset with the new columns from pandas dataframe.
+
+        Parameters
+        ----------
+        df:
+            Dataframe with the new columns in wide etna format
+            If columns with the same names already exist in the dataset, there values will be updated
+        regressors:
+            List of regressors in the passed dataframe
+        """
+        columns_in_dataset = self.columns.get_level_values("feature")
+        new_columns = df.columns.get_level_values("feature")
+
+        columns_to_update = list(set(columns_in_dataset) & set(new_columns))
+        self.df.loc[:, self.idx[:, columns_to_update]] = df.loc[:, self.idx[:, columns_to_update]]
+
+        columns_to_add = list(set(new_columns) - set(columns_in_dataset))
+        self.df = self.df.join(df.loc[:, self.idx[:, columns_to_add]]).sort_index(axis=1)
+
+        if regressors is not None:
+            self._regressors = list(set(self._regressors) | set(regressors))
```

Comment:
With this regressors there is a problem that they aren't saved in `df_exog` and can't be used in the future. It can be not obvious for a user, we should write about it in documentation.
E.g. user want to add a new regressors, he tries to use this function with columns extended into the future and fails with it because this method probably doesn't work correctly in this case.

Reply:
Add comment in doc